### PR TITLE
lib: Use language-specific heredoc delimiters for C code

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- I realised these weren't part of #195129 because my batching in that PR covered `Formula/{g,h,j,k,l}/` which didn't include `Formula/lib/`.

### CODE REVIEW

File: Formula/lib/libtool.rb [hunk 1]
```diff
@@ -47,22 +47,22 @@ def install
...
   def caveats
     on_macos do
-      <<~EOS
+      <<~C
         All commands have been installed with the prefix "g".
         If you need to use these commands with their normal names, you
         can add a "gnubin" directory to your PATH from your bashrc like:
           PATH="#{opt_libexec}/gnubin:$PATH"
-      EOS
+      C
```
Comment (carlocab) [inline Formula/lib/libtool.rb:55]: Don't think we want this 😅

File: Formula/lib/libxslt.rb [hunk 2]
```diff
@@ -58,21 +58,21 @@ def install
   end
 
   def caveats
-    <<~EOS
+    <<~C
       To allow the nokogiri gem to link against this libxslt run:
         gem install nokogiri -- --with-xslt-dir=#{opt_prefix}
-    EOS
+    C
```
Comment (carlocab) [inline Formula/lib/libxslt.rb:64]: Or this!

**Review by carlocab — APPROVED** [review]

File: Formula/lib/libxmi.rb [hunk 3]
```diff
@@ -93,7 +93,7 @@ def install
       000000000000000000000200000000000000000000000000000\n000000000000000000000020000000000000000000000000000
       000000000000000000000002000000000000000000000000000\n000000000000000000000000003000000000000000000000000
       000000000000000000000000000333000000000000000000000\n000000000000000000000000000000004444000000000000000
-    EOS
+    C
```
Comment (stefanb) [inline Formula/lib/libxmi.rb:96]: This also not C code.

File: Formula/lib/libsvg.rb [hunk 4]
```diff
@@ -51,11 +51,11 @@ def install
   end
 
   test do
-    (testpath/"test.svg").write <<~EOS
+    (testpath/"test.svg").write <<~C
```
Comment (stefanb) [inline Formula/lib/libsvg.rb:54]: another one

File: Formula/lib/libtool.rb (hunk 1 above)
Comment (issyl0) [reply Formula/lib/libtool.rb:55]: My script went rogue!

**Comment by issyl0** [comment]
@carlocab @stefanb I am grateful for your 🔍 reviews! These should be fixed now, and I have tweaked my script to not `sed` over the whole file, just the range where it finds the `.#{extension}").write <<~EOS`.

**Review by carlocab — APPROVED** [review]

**Pull request merged**